.github/workflows/test-poetry-build.yml

```diff
@@ -0,0 +1,53 @@
+name: Poetry Test
+on:
+  # Trigger the workflow on push or pull request,
+  # but only for the main branch
+  push:
+    branches:
+      - master
+      - 'releases/**'
+  pull_request:
+    branches:
+      - master
+      - 'releases/**'
+jobs:
+  poetry-build-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Install Poetry
+        run: |
+          curl -sSL https://install.python-poetry.org | python -
+          echo "$HOME/.poetry/bin" >> $GITHUB_PATH
+      - name: Create foo package
+        run: |
+          mkdir foo
+          cat <<EOF > foo/pyproject.toml
+          [tool.poetry]
+          name = "foo"
+          version = "1.0.0"
+          authors = ["skypilot-bot"]
+          description = ""
+
+          [tool.poetry.dependencies]
+          python = "3.10.x"
+
+          [tool.poetry.group.dev.dependencies]
+          skypilot = {git = "https://github.com/skypilot-org/skypilot.git", branch = "${{ github.head_ref }}"}
+
+          [build-system]
+          requires = ["poetry-core"]
+          build-backend = "poetry.core.masonry.api"
+          
+          EOF
+
+      - name: Check poetry lock time
+        run: |
+          cd foo
+          poetry lock --no-update
+        timeout-minutes: 1
```

Comment:
Curious - is 1 min sufficient? Running `poetry lock --no-update` on my M2 Max Mac took 57s, which is cutting close to the timeout. Not sure how long it takes on a github action...

```
(base) ➜  foo poetry lock --no-update
Creating virtualenv foo-tFXSep41-py3.9 in /Users/romilb/Library/Caches/pypoetry/virtualenvs
Updating dependencies
Resolving dependencies... Downloading https://files.pythonhosted.org/packages/1b/43/0b89c6ba8b3b23de6847164faae88
Resolving dependencies... Downloading https://files.pythonhosted.org/packages/e0/4f/4474bda990ee740a020cbc3eb2719
Resolving dependencies... Downloading https://files.pythonhosted.org/packages/a6/60/b91acaa7995bfcd72f1739ea2b0f5
Resolving dependencies... Downloading https://files.pythonhosted.org/packages/a6/60/b91acaa7995bfcd72f1739ea2b0f5
Resolving dependencies... Downloading https://files.pythonhosted.org/packages/62/d5/5f610ebe421e85889f2e55e33b7f9
Resolving dependencies... Downloading https://files.pythonhosted.org/packages/d9/5a/e7c31adbe875f2abbb91bd84cf2dc
Resolving dependencies... Downloading https://files.pythonhosted.org/packages/96/06/4beb652c0fe16834032e54f095644
Resolving dependencies... Downloading https://files.pythonhosted.org/packages/43/a0/9ba967fdbd55293bacfc1507f58e3
Resolving dependencies... Downloading https://files.pythonhosted.org/packages/d5/50/8aedb5ff1460e7c8527af15c63261
Resolving dependencies... Downloading https://files.pythonhosted.org/packages/d5/50/8aedb5ff1460e7c8527af15c63261
Resolving dependencies... Downloading https://files.pythonhosted.org/packages/d5/50/8aedb5ff1460e7c8527af15c63261
Resolving dependencies... Downloading https://files.pythonhosted.org/packages/b4/ff/9f10044630e55eb6fca1c32a00710
Resolving dependencies... (57.8s)

Writing lock file
```

Reply:
Hmm, interesting. It seems to be taking 37s on [github CI](https://github.com/skypilot-org/skypilot/actions/runs/6080962115/job/16495751653?pr=2515), which might be because it has better network. 

I can increase the timeout in CI to 2 mins. : )

Reply:
2 min sounds good!